File: Easy/valid-mountain-array.java
class Solution {        // 1ms
    public boolean validMountainArray(int[] arr) {
        int n = arr.length;
        int i = 0;
        
        for(; i + 1 < n && arr[i] < arr[i + 1]; i++);
        if(i == 0 || i + 1 == n) return false;
        for(; i + 1 < n && arr[i] > arr[i + 1]; i++);
        return i == n - 1;
    }
}

File: Easy/valid-mountain-array.py
class Solution:     # 317ms
    def validMountainArray(self, arr: List[int]) -> bool:
        if len(arr) < 3 or arr[0] > arr[1]:
            return False 
        
        i = 1
        while i < len(arr) - 1 and arr[i - 1] < arr[i]:
            i += 1
        while i < len(arr) and arr[i - 1] > arr[i]:
            i += 1
        return i == len(arr)